exercises/java/src/main/java/com/adaptionsoft/games/uglytrivia/Game.java

```diff
@@ -1,169 +1,135 @@
 package com.adaptionsoft.games.uglytrivia;
 
+import com.adaptionsoft.games.Board;
+import com.adaptionsoft.games.categories.Category;
+
 import java.util.ArrayList;
-import java.util.LinkedList;
+import java.util.Arrays;
+import java.util.Map;
 
 public class Game {
-    ArrayList players = new ArrayList();
-    int[] places = new int[6];
-    int[] purses  = new int[6];
-    boolean[] inPenaltyBox  = new boolean[6];
-    
-    LinkedList popQuestions = new LinkedList();
-    LinkedList scienceQuestions = new LinkedList();
-    LinkedList sportsQuestions = new LinkedList();
-    LinkedList rockQuestions = new LinkedList();
-    
-    int currentPlayer = 0;
-    boolean isGettingOutOfPenaltyBox;
-    
-    public  Game(){
-    	for (int i = 0; i < 50; i++) {
-			popQuestions.addLast("Pop Question " + i);
-			scienceQuestions.addLast(("Science Question " + i));
-			sportsQuestions.addLast(("Sports Question " + i));
-			rockQuestions.addLast(createRockQuestion(i));
-    	}
+    private Map<Integer, Category> boardPositions;
+    private ArrayList<String> playerNames = new ArrayList<>();
+    private static final int MAX_AMOUNT_OF_PLAYERS = 6;
+    private int[] locationOfPlayerOnBoard = new int[MAX_AMOUNT_OF_PLAYERS];
+    private int[] playerPurses = new int[MAX_AMOUNT_OF_PLAYERS];
+    private boolean[] isPlayerInPenaltyBox = new boolean[MAX_AMOUNT_OF_PLAYERS];
+    private boolean isGettingOutOfPenaltyBox;
+
+    private int currentPlayer = 0;
+    private String currentPlayerName;
+
+    public Game(Board board) {
+        this.boardPositions = board.createBoard();
+    }
+
+    public void addPlayersToGame(String... players) {
+        Arrays.stream(players).forEach(playerName -> {
+            playerNames.add(playerName);

```

Comment:
Feels like there's a Player constructor call hiding in here somewhere, though we're several steps aways from pull that out I think.  Interesting use of streams, though I wonder whether that will make a difference to how easy it will be to do other refactors?